src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs

```diff
@@ -382,11 +389,170 @@ public void SendPacketsElement_FileLargeCount_Throws(SocketImplementationType ty
         {
             // Length is validated on Send
             SendPackets(type, new SendPacketsElement(TestFileName, 5, 10000), SocketError.InvalidArgument, 0);
+            SendPackets(type, new SendPacketsElement(TestFileName, 5L, 10000), SocketError.InvalidArgument, 0);
         }
 
         #endregion Files
 
+        #region FileStreams
+
+        [Theory]
+        [InlineData(SocketImplementationType.APM)]
+        [InlineData(SocketImplementationType.Async)]
+        public void SendPacketsElement_FileStream_Success(SocketImplementationType type)
+        {
+            using (var stream = new FileStream(TestFileName, FileMode.Open, FileAccess.Read))
+            {
+                stream.Seek(s_testFileSize / 2, SeekOrigin.Begin);
+                SendPackets(type, new SendPacketsElement(stream), s_testFileSize); // Whole File
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    Assert.Equal(s_testFileSize / 2, stream.Position);
```

Comment:
Does this Assert pass on Windows when the FileStream has `FileOptions.Asynchronous`?

Reply:
Yes this Assert pass on Windows for async FileStream.

Reply:
I think it makes sense to make SendPackets throw when used with non async FileStreams. Then we can remove the workaround in SocketAsyncEventArgs.Windows.cs and remove these special cases from the tests. It has consistent cross-platform behavior and I don't see any downsides.

@geoffkizer @stephentoub @mikedn @AlexRadch wdyt?

Reply:
I looks reasonable for me because for non async FileStreams application can not use such FileStreams while SendPacketsAsync works.